content/en/docs/concepts/services-networking/service.md

```diff
@@ -827,434 +628,268 @@ metadata:
-        service.beta.kubernetes.io/aws-load-balancer-extra-security-groups: "sg-53fae93f,sg-42efd82e"
-
-        # A comma separated list of key-value pairs which are used
-        # to select the target nodes for the load balancer
-        service.beta.kubernetes.io/aws-load-balancer-target-node-labels: "ingress-gw,gw-name=public-api"
-```
+Sometimes you don't need load-balancing and a single Service IP.  In
+this case, you can create what are termed _headless Services_, by explicitly
+specifying `"None"` for the cluster IP address (`.spec.clusterIP`).
 
-#### Network Load Balancer support on AWS {#aws-nlb-support}
+You can use a headless Service to interface with other service discovery mechanisms,
+without being tied to Kubernetes' implementation.
 
-{{< feature-state for_k8s_version="v1.15" state="beta" >}}
+For headless Services, a cluster IP is not allocated, kube-proxy does not handle
+these Services, and there is no load balancing or proxying done by the platform
+for them. How DNS is automatically configured depends on whether the Service has
+selectors defined:
 
-To use a Network Load Balancer on AWS, use the annotation `service.beta.kubernetes.io/aws-load-balancer-type` with the value set to `nlb`.
+### With selectors
 
-```yaml
-    metadata:
-      name: my-service
-      annotations:
-        service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
-```
+For headless Services that define selectors, the Kubernetes control plane creates
+EndpointSlice objects in the Kubernetes API, and modifies the DNS configuration to return
+A records (IP addresses) that point directly to the Pods backing the Service.
 
-{{< note >}}
-NLB only works with certain instance classes; see the
-[AWS documentation](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-register-targets.html#register-deregister-targets)
-on Elastic Load Balancing for a list of supported instance types.
-{{< /note >}}
+### Without selectors
 
-Unlike Classic Elastic Load Balancers, Network Load Balancers (NLBs) forward the
-client's IP address through to the node. If a Service's `.spec.externalTrafficPolicy`
-is set to `Cluster`, the client's IP address is not propagated to the end
-Pods.
+For headless Services that do not define selectors, the control plane does
+not create EndpointSlice objects. However, the DNS system looks for and configures
+either:
 
-By setting `.spec.externalTrafficPolicy` to `Local`, the client IP addresses is
-propagated to the end Pods, but this could result in uneven distribution of
-traffic. Nodes without any Pods for a particular LoadBalancer Service will fail
-the NLB Target Group's health check on the auto-assigned
-`.spec.healthCheckNodePort` and not receive any traffic.
+* DNS CNAME records for [`type: ExternalName`](#externalname) Services.
+* DNS A records for any Endpoints that share a name with the Service, for all
+  types other than `ExternalName`.
```

Comment:
@chrisohaver do you mind review this DNS part?

Reply:
OK.  I've commented with a suggested a change.